include/seqan3/argument_parser/validators.hpp

```diff
@@ -0,0 +1,386 @@
+    { validator(value) };
+    { validator(value_vector) };
+    { validator.get_help_page_message() } -> std::string;
+};
+
+/*!\brief A validator that checks whether a number is inside a given range.
+ *
+ * \tparam option_value_type Must be a (container of) integral type(s).
+ *
+ * \details
+ * On construction, the validator must receive a maximum and a minimum number.
+ * The struct than acts as a functor, that throws a seqan3::parser_invalid_argument
+ * exception whenever a given value does not lie inside the given min/max range.
+ *
+ *```cpp
+ *     int main(int argc, char ** argv)
+ *     {
+ *          seqan3::argument_parser myparser(argc, argv); // initialize
+ *
+ *          int myint;
+ *          seqan3::range_validator my_validator{2, 10};
+ *
+ *          myparser.add_option(myint,'i',"integer","Give me a number.",
+ *                              seqan3::option_spec::DEFAULT, my_validator);
+ *
+ *          // an exception will be thrown if the user specifies an integer
+ *          // that is not in range [2,10] (e.g. "./test_app -i 15")
+ *          try
+ *          {
+ *               myparser.parse();
+ *          }
+ *          catch (seqan3::parser_invalid_argument const & ext) // the user did something wrong
+ *          {
+ *               std::cerr << "[PARSER ERROR] " << ext.what(); // customize your error message
+ *               return -1;
+ *          }
+ *          catch (seqan3::parser_interruption const &) // expected behaviour on special requests (e.g. `--help`)
+ *          {
+ *               return 0;
+ *          }
+ *
+ *          std::cout << "integer given by user passed validation: " << myint << endl;
+ *          return 0;
+ *     }
+ *```
+ */
+template <typename option_value_type>
+//!\cond
+    requires integral_concept<option_value_type> || integral_concept<typename option_value_type::value_type>
+//!\cond
+class range_validator
+{
+public:
+    using value_type = option_value_type; //! Type of values that are tested by validator
+
+    /*!\brief The constructor.
+     *
+     * \param min_ Minimum set for the range to test.
+     * \param max_ Maximum set for the range to test.
+     */
+    range_validator(option_value_type const min_, option_value_type const max_) :
+        min{min_}, max{max_}
+    {}
+
+    /*!\brief Tests whether cmp lies inside [min,max].
+     * \throws parser_invalid_argument
+     */
+    void operator()(option_value_type const & cmp) const
+    {
+        if (!((cmp <= max) && (cmp >= min)))
+            throw parser_invalid_argument(detail::paste("Validation Failed - Value ", cmp,
+                                                        " is not in range [", min, ",", max, "]."));
+    }
+
+    //!\brief Tests whether all values of v lie inside [min,max].
+    template <typename container_type>
+    //!\cond
+        requires container_concept<container_type> &&
+                 std::is_same<typename container_type::value_type, option_value_type>::value
```

Comment:
please use `range_value_type_t<container_type>` instead

Reply:
as discussed offline, I'll leave this be.